### DISCUSSION TITLE
TurnKey LXC Template Scripts

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#1742] [Announcements]
Primitive [scripts](https://github.com/tteck/Proxmox/tree/main/turnkey) to install [TurnKey LXC templates](https://www.turnkeylinux.org/) using basic settings.

**Resource and network settings are adjustable post LXC creation.**

More can be added on request.

**I would like to keep the TurnKey scripts as simple as possible (single file) with few options (if any).** 

The script creates a `*.creds` file in the Proxmox root directory with the password of the newly created TurnKey LXC.
To view the password:
```
cat turnkey-<name>.creds
```

**Comment by tteck** [comment]
**When encountering** 
![Screenshot 2023-08-29 2 19 54 PM](https://github.com/tteck/Proxmox/assets/21342670/0cf5c926-70a3-4c96-801a-9c83255944ac)
**enter the LXC IP**
![Screenshot 2023-08-29 2 20 46 PM](https://github.com/tteck/Proxmox/assets/21342670/0c288abd-9a81-4a0f-8b76-c36cc7a4c420)

**Comment by elRadix** [comment]
I'm not succeeding of installing these new scripts,. tried the fileserver and observium and still getting the error message on the screenshot below. Any idea how to fix this?
![Screenshot_20230830_224259_Chrome](https://github.com/tteck/Proxmox/assets/6258894/47641521-71f6-4b79-9b90-2102c6755c01)

**Reply by tteck** [reply]
Try fileserver again.

**Reply by tteck** [reply]
It appears that your network might not be receptive to the code snippet: `pct exec $CTID ip route get 8.8.8.8 | sed -n '/src/{s/.*src *\([^ ]*\).*/\1/p;q}'` in the script.

**Reply by tteck** [reply]
In the absence of a displayed IP address, you might face uncertainty regarding the input to provide when prompted with the question: "Enter the domain to serve ***" in some applications.

**Reply by tteck** [reply]
Alternatively, the conventional network settings `-net0 name=eth0,bridge=vmbr0,ip=dhcp` might not function as expected within your system.

**Reply by tteck** [reply]
Is access to Google's public DNS servers restricted either by your service or within your country?

**Reply by tteck** [reply]
@elRadix Could you offer any updates regarding your issue?

**Reply by elRadix** [reply]
See below, regarding DNS i'm running 2 adguard LXC so no issue there also in proxmox  I have cloudflare as fallback
![image](https://github.com/tteck/Proxmox/assets/6258894/4e4a6943-798d-4a57-8424-c81ada1cc959)

**Comment by elRadix** [comment]
It works now with the fileserver image but still not with observium image, what's the diference within those 2 images?

**Reply by tteck** [reply]
I temporarily removed the line `pct exec $CTID ip route get 8.8.8.8 | sed -n '/src/{s/.*src *\([^ ]*\).*/\1/p;q}'` from the code in the 'fileserver' script. Once you identify and resolve the problem with that particular code snippet, everything should start working as intended.

While you're in the Proxmox shell, you can experiment with different versions of the code snippet. Instead of using '$CTID', replace it with the ID of a running container. This will help you check if the snippet can give you the container's IP address.

```
pct exec $CTID ip route get 8.8.8.8 | sed -n '/src/{s/.*src *\([^ ]*\).*/\1/p;q}'
```
```
pct exec $CTID ip route get 1.1.1.1 | sed -n '/src/{s/.*src *\([^ ]*\).*/\1/p;q}'
```

**Reply by elRadix** [reply]
this the result:
![image](https://github.com/tteck/Proxmox/assets/6258894/c4292a36-f643-47b8-a663-d1a01b95e7cc)

**Reply by tteck** [reply]
That would suggest that the code is functioning as expected. Another potential factor that could be causing the problem is the 5-second delay I've implemented after starting the LXC to ensure the network has sufficient time to establish a connection. However, in your situation, it's possible that this delay might not be sufficient.

Whenever you have a moment, I'd be interested in exploring a few different methods to address this issue. Please let me know when you're available to proceed, and we can work through these options together.

**Reply by tteck** [reply]
When you have the opportunity, could you please attempt to create a new TurnKey Observium LXC? Keep in mind that there have been changes to the code.

**Reply by kotovasia5120** [reply]
I have the same problem. My router does not have time to register a new device and a connection error appears. Each time a new ip and mac, so the router registers a different device

**Reply by tteck** [reply]
Each TurnKey script includes the following code snippet. It verifies whether the container has an IP address. If not, it pauses for 5 seconds and rechecks. This process repeats for a total of 5 attempts before the script proceeds, even if the container still lacks a known IP address. However, it doesn't provide a method to alter the MAC address of the running container.
```
max_attempts=5
attempt=1
IP=""
while [[ $attempt -le $max_attempts ]]; do
  IP=$(pct exec $CTID ip route get 8.8.8.8 | sed -n '/src/{s/.*src *\([^ ]*\).*/\1/p;q}')
  if [[ -n $IP ]]; then
    break
  else
    warn "Attempt $attempt: IP address not found. Pausing for 5 seconds..."
    sleep 5
    ((attempt++))
  fi
done

if [[ -z $IP ]]; then
  warn "Maximum number of attempts reached. IP address not found."
  IP="NOT FOUND"
fi
```

**Reply by kotovasia5120** [reply]
Reloading proxmox most often helps, I think because resources are spent on startup and everything is slower.
![image](https://github.com/tteck/Proxmox/assets/76496357/53c61119-2b94-482a-967d-bbe2fa3a140d)

**Reply by tteck** [reply]
My suggestion is for you to download the script, delete the specified code snippet, and run the script locally.

**Reply by tteck** [reply]
@kotovasia5120 I just pushed this [commit](https://github.com/tteck/Proxmox/commit/3b1560fa63a85ba121995f0bbfe98fe455d2d144) to use an alternative method for acquiring the IP address in the TurnKey scripts.

**Please inform me if the TurnKey scripts are now functioning properly for you.**

**Comment by Vakrehus** [comment]
### Observium:

For all users who cannot add a device because it cannot be pinged.

The problem can be solved by execute `chmod +s /usr/bin/fping` in Observium-LXC.

**Comment by Dulanic** [comment]
I've been trying to get nextcloud turnkey working and it just wont get any dns responses. I set it to DHCP because all my other containers work fine that way, and I can assign them a reserved IP. They cannot connect to any external sites at all. So I tried static, /w 1.1.1.1 as DNS, and still same. For some reason its setting my broacast to 192.168.3 instead of 192.168.2? Possible reason? Im really not sure. The webui etc... is responsive, it just has 0 net connection.

![image](https://github.com/tteck/Proxmox/assets/2948206/8747c7e4-dce4-41f4-a6af-5cd948e4162c)

Edit: OK I figured it out, for some reason turnkey wanted to set my DNS to 100.100.100.100, I edited /etc/resolv.conf and fixed it. So weird considring I manually changed it with the menu, but it didn't take.

**Reply by tteck** [reply]
You're using a `/22` CIDR notation, is that correct?

**Reply by Dulanic** [reply]
> You're using a `/22` CIDR notation, is that correct?

I didnt actually set that, i typed it in the turnkey network settings. I can fix it by editing /etc/resolv.cnf but on container restart, its back to what it was. 
![image](https://github.com/tteck/Proxmox/assets/2948206/df5ab798-57c5-4cc3-8c75-c04f14fecec3)

Edit: I also tried proxmox network setting, and it isnt working there either. It still sets /etc/resolv.conf to 100.100.100.100.
![image](https://github.com/tteck/Proxmox/assets/2948206/cb03c9ec-10cb-4ed7-82fe-10fbde74c648)

Edit: Fixed it,  had to change the DNS server from host to what I wanted manually. Weird that this is the only container /w that problem.

**Reply by tteck** [reply]
I haven't utilized TurnKey network settings before, should you add the `/24` CIDR notation in the TurnKey IP setting?

**Reply by tteck** [reply]
What was the reason for your utilization of TurnKey network settings?

**Reply by Dulanic** [reply]
Real reason? I'm new to lxc and proxmox. I was used to docker.

**Reply by tteck** [reply]
I believe it would be a good idea to remove the existing container and create a new one, exclusively utilizing the LXC network settings.

**Comment by tteck** [comment]
I've created a `turnkey.sh` script designed to manage the creation of TurnKey LXC containers.

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/turnkey/turnkey.sh)"
```
This has streamlined the process and eliminated the need for maintaining 13 separate scripts.

**Reply by swilkey** [reply]
Unfortunately it didn't work for me.  I had to remove all the set +e and set -e lines because without them the script would install the container and then delete it before starting it.  In reality it had successfully created the container, obtained an IP address for it, and set the username and password for it - all it needed to do was to tell me that and not to delete them.  With those adjustments it worked.

**Reply by tteck** [reply]
I _think_ that I have the issue resolved. Would you be so kind and create a new LXC to test?

**Comment by swilkey** [comment]
Thanks.  It got further (see below).  This time only 1 unknown failure 
occurred instead of two, but with the one that occurred it destroyed the 
newly created container again.

  ______              __ __           __   _  _______
/_  __/_ _________  / //_/__ __ __  / /  | |/_/ ___/
  / / / // / __/ _ \/ ,< / -_) // / / /___>  </ /__
/_/  \_,_/_/ /_//_/_/|_|\__/\_, / /____/_/|_|\___/
                            /___/
[INFO] Using 'local' for template storage.
[INFO] Using 'local' for container storage.
Updating LXC template list...
Creating LXC container...
Starting LXC Container...
[ERROR] ***@***.*** Unknown failure occured.
***@***.***:~#


------ Original Message ------
From: "tteckster" ***@***.***>
To: "tteck/Proxmox" ***@***.***>
Cc: "swilkey" ***@***.***>; "Comment" 
***@***.***>
Sent: 4/12/2023 9:59:18 PM
Subject: Re: [tteck/Proxmox] TurnKey LXC Template Scripts (Discussion 
#1742)


I think that I have the issue resolved. Would you be so kind and create 
a new LXC to test?

—
Reply to this email directly, view it on GitHub 
<https://github.com/tteck/Proxmox/discussions/1742#discussioncomment-7751843>, 
or unsubscribe 
<https://github.com/notifications/unsubscribe-auth/AB5UXTEDVUUKNCZ5JDKRSW3YHWUINAVCNFSM6AAAAAA4DDCVYWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TONJRHA2DG>.
You are receiving this because you commented.Message ID: 
***@***.***>

**Comment by tteck** [comment]
The script runs without any problems from this end. 

```
 ______              __ __           __   _  _______
/_  __/_ _________  / //_/__ __ __  / /  | |/_/ ___/
 / / / // / __/ _ \/ ,< / -_) // / / /___>  </ /__
/_/  \_,_/_/ /_//_/_/|_|\__/\_, / /____/_/|_|\___/
                           /___/
[INFO] Using 'local' for template storage.
[INFO] Using 'local-lvm' for container storage.
Updating LXC template list...
Creating LXC container...
Starting LXC Container...

 ______              __ __           __   _  _______
/_  __/_ _________  / //_/__ __ __  / /  | |/_/ ___/
 / / / // / __/ _ \/ ,< / -_) // / / /___>  </ /__
/_/  \_,_/_/ /_//_/_/|_|\__/\_, / /____/_/|_|\___/
                           /___/

[INFO] LXC container '102' was successfully created, and its IP address is 192.168.86.253.

[INFO] Proceed to the LXC console to complete the setup.

[INFO] login: root
[INFO] password: 5Fsb9xb/wQ0=
```

**Comment by swilkey** [comment]
If you can tell me how to run your script with debugging turned on I can 
give you the output of it step by step and you can see where it is 
failing.

------ Original Message ------
From: "tteckster" ***@***.***>
To: "tteck/Proxmox" ***@***.***>
Cc: "swilkey" ***@***.***>; "Comment" 
***@***.***>
Sent: 5/12/2023 7:22:22 AM
Subject: Re: [tteck/Proxmox] TurnKey LXC Template Scripts (Discussion 
#1742)


The script runs without any problems from this end.

  ______              __ __           __   _  _______
/_  __/_ _________  / //_/__ __ __  / /  | |/_/ ___/
  / / / // / __/ _ \/ ,< / -_) // / / /___>  </ /__
/_/  \_,_/_/ /_//_/_/|_|\__/\_, / /____/_/|_|\___/
                            /___/
[INFO] Using 'local' for template storage.
[INFO] Using 'local-lvm' for container storage.
Updating LXC template list...
Creating LXC container...
Starting LXC Container...

  ______              __ __           __   _  _______
/_  __/_ _________  / //_/__ __ __  / /  | |/_/ ___/
  / / / // / __/ _ \/ ,< / -_) // / / /___>  </ /__
/_/  \_,_/_/ /_//_/_/|_|\__/\_, / /____/_/|_|\___/
                            /___/

[INFO] LXC container '102' was successfully created, and its IP address is 192.168.86.253.

[INFO] Proceed to the LXC console to complete the setup.

[INFO] login: root
[INFO] password: 5Fsb9xb/wQ0=

—
Reply to this email directly, view it on GitHub 
<https://github.com/tteck/Proxmox/discussions/1742#discussioncomment-7757305>, 
or unsubscribe 
<https://github.com/notifications/unsubscribe-auth/AB5UXTFGOKN7RUHOTU5FCGDYHYWH5AVCNFSM6AAAAAA4DDCVYWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TONJXGMYDK>.
You are receiving this because you commented.Message ID: 
***@***.***>

**Reply by tteck** [reply]
Use the set -x option
```
bash -c "set -x; $(wget -qLO - https://github.com/tteck/Proxmox/raw/main/turnkey/turnkey.sh)"
```

**Comment by swilkey** [comment]
Thanks.  Hopefully this is helpful.

+ cat
  ______              __ __           __   _  _______
/_  __/_ _________  / //_/__ __ __  / /  | |/_/ ___/
  / / / // / __/ _ \/ ,< / -_) // / / /___>  </ /__
/_/  \_,_/_/ /_//_/_/|_|\__/\_, / /____/_/|_|\___/
                            /___/
+ whiptail --backtitle 'Proxmox VE Helper Scripts' --title 'TurnKey 
LXCs' --yesno 'This will allow for the creation of one of the many 
TurnKey LXC Containers. Proceed?' 10 68
+ TURNKEY_MENU=()
+ MSG_MAX_LENGTH=0
+ read -r TAG ITEM
++ cat
+ OFFSET=2
+ (( 7 + OFFSET > MSG_MAX_LENGTH ))
+ MSG_MAX_LENGTH=7+OFFSET
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 9 + OFFSET > MSG_MAX_LENGTH ))
+ MSG_MAX_LENGTH=9+OFFSET
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 4 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 14 + OFFSET > MSG_MAX_LENGTH ))
+ MSG_MAX_LENGTH=14+OFFSET
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 11 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 7 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 11 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 5 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 6 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 13 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 12 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 9 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 9 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 4 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 7 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 8 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 5 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 14 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 9 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 9 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
+ OFFSET=2
+ (( 10 + OFFSET > MSG_MAX_LENGTH ))
+ TURNKEY_MENU+=("$TAG" "$ITEM " "OFF")
+ read -r TAG ITEM
++ tr -d '"'
++ whiptail --backtitle 'Proxmox VE Helper Scripts' --title 'TurnKey 
LXCs' --radiolist '\nSelect a TurnKey LXC to create:\n' 16 74 6 ansible 
'Ansible ' OFF bookstack 'BookStack ' OFF core 'Core ' OFF 
faveo-helpdesk 'Faveo Helpdesk ' OFF fileserver 'File Server ' OFF 
gallery 'Gallery ' OFF gameserver 'Game Server ' OFF gitea 'Gitea ' OFF 
gitlab 'GitLab ' OFF invoice-ninja 'Invoice Ninja ' OFF mediaserver 
'Media Server ' OFF nextcloud 'Nextcloud ' OFF observium 'Observium ' 
OFF odoo 'Odoo ' OFF openvpn 'OpenVPN ' OFF owncloud 'ownCloud ' OFF 
phpbb 'phpBB ' OFF torrentserver 'Torrent Server ' OFF wireguard 
'WireGuard ' OFF wordpress 'Wordpress ' OFF zoneminder 'ZoneMinder ' OFF
+ turnkey=nextcloud
+ '[' -z nextcloud ']'
++ openssl rand -base64 8
+ PASS=gNE05GvOiBQ=
++ pvesh get /cluster/nextid
+ CTID=105
+ PCT_OPTIONS='
     -features keyctl=1,nesting=1
     -hostname turnkey-nextcloud
     -tags proxmox-helper-scripts
     -onboot 1
     -cores 2
     -memory 2048
     -password gNE05GvOiBQ=
     -net0 name=eth0,bridge=vmbr0,ip=dhcp
     -unprivileged 1
   '
+ DEFAULT_PCT_OPTIONS=(-arch $(dpkg --print-architecture))
++ dpkg --print-architecture
++ select_storage template
++ local CLASS=template
++ local CONTENT
++ local CONTENT_LABEL
++ case $CLASS in
++ CONTENT=vztmpl
++ CONTENT_LABEL='Container template'
++ local -a MENU
++ read -r line
+++ pvesm status -content vztmpl
+++ awk 'NR>1'
+++ echo local dir active 959318096 457403764 453109932 47.68%
+++ awk '{print $1}'
++ local TAG=local
+++ echo local dir active 959318096 457403764 453109932 47.68%
+++ awk '{printf "%-10s", $2}'
++ local 'TYPE=dir       '
+++ echo local dir active 959318096 457403764 453109932 47.68%
+++ numfmt --field 4-6 --from-unit=K --to=iec --format %.2f
+++ awk '{printf( "%9sB", $6)}'
++ local 'FREE=  422.00GB'
++ local 'ITEM=  Type: dir        Free:   422.00GB '
++ local OFFSET=2
++ [[ 38 -gt 14+OFFSET ]]
++ local MSG_MAX_LENGTH=38
++ MENU+=("$TAG" "$ITEM" "OFF")
++ read -r line
++ '[' 1 -eq 0 ']'
++ '[' 1 -eq 1 ']'
++ printf local
+ TEMPLATE_STORAGE=local
+ info 'Using '\''local'\'' for template storage.'
+ local 'REASON=Using '\''local'\'' for template storage.'
+ local 'FLAG=\e[36m[INFO]\e[39m'
+ msg '\e[36m[INFO]\e[39m Using '\''local'\'' for template storage.'
+ local 'TEXT=\e[36m[INFO]\e[39m Using '\''local'\'' for template 
storage.'
+ echo -e '\e[36m[INFO]\e[39m Using '\''local'\'' for template storage.'
[INFO] Using 'local' for template storage.
++ select_storage container
++ local CLASS=container
++ local CONTENT
++ local CONTENT_LABEL
++ case $CLASS in
++ CONTENT=rootdir
++ CONTENT_LABEL=Container
++ local -a MENU
++ read -r line
+++ pvesm status -content rootdir
+++ awk 'NR>1'
+++ echo local dir active 959318096 457403764 453109932 47.68%
+++ awk '{print $1}'
++ local TAG=local
+++ echo local dir active 959318096 457403764 453109932 47.68%
+++ awk '{printf "%-10s", $2}'
++ local 'TYPE=dir       '
+++ echo local dir active 959318096 457403764 453109932 47.68%
+++ numfmt --field 4-6 --from-unit=K --to=iec --format %.2f
+++ awk '{printf( "%9sB", $6)}'
++ local 'FREE=  422.00GB'
++ local 'ITEM=  Type: dir        Free:   422.00GB '
++ local OFFSET=2
++ [[ 38 -gt 14+OFFSET ]]
++ local MSG_MAX_LENGTH=38
++ MENU+=("$TAG" "$ITEM" "OFF")
++ read -r line
++ '[' 1 -eq 0 ']'
++ '[' 1 -eq 1 ']'
++ printf local
+ CONTAINER_STORAGE=local
+ info 'Using '\''local'\'' for container storage.'
+ local 'REASON=Using '\''local'\'' for container storage.'
+ local 'FLAG=\e[36m[INFO]\e[39m'
+ msg '\e[36m[INFO]\e[39m Using '\''local'\'' for container storage.'
+ local 'TEXT=\e[36m[INFO]\e[39m Using '\''local'\'' for container 
storage.'
+ echo -e '\e[36m[INFO]\e[39m Using '\''local'\'' for container 
storage.'
[INFO] Using 'local' for container storage.
+ msg 'Updating LXC template list...'
+ local 'TEXT=Updating LXC template list...'
+ echo -e 'Updating LXC template list...'
Updating LXC template list...
+ pveam update
+ mapfile -t TEMPLATES
++ pveam available -section turnkeylinux
++ awk -v turnkey=nextcloud '$0 ~ turnkey {print $2}'
++ sort -t - -k 2 -V
+ '[' 1 -gt 0 ']'
+ TEMPLATE=debian-12-turnkey-nextcloud_18.0-1_amd64.tar.gz
+ pveam list local
+ grep -q debian-12-turnkey-nextcloud_18.0-1_amd64.tar.gz
+ PCT_OPTIONS=(${PCT_OPTIONS[@]:-${DEFAULT_PCT_OPTIONS[@]}})
+ [[  -features keyctl=1,nesting=1 -hostname turnkey-nextcloud -tags 
proxmox-helper-scripts -onboot 1 -cores 2 -memory 2048 -password 
gNE05GvOiBQ= -net0 name=eth0,bridge=vmbr0,ip=dhcp -unprivileged 1  =~  
-rootfs  ]]
+ PCT_OPTIONS+=(-rootfs $CONTAINER_STORAGE:${PCT_DISK_SIZE:-8})
+ msg 'Creating LXC container...'
+ local 'TEXT=Creating LXC container...'
+ echo -e 'Creating LXC container...'
Creating LXC container...
+ pct create 105 
local:vztmpl/debian-12-turnkey-nextcloud_18.0-1_amd64.tar.gz -features 
keyctl=1,nesting=1 -hostname turnkey-nextcloud -tags 
proxmox-helper-scripts -onboot 1 -cores 2 -memory 2048 -password 
gNE05GvOiBQ= -net0 name=eth0,bridge=vmbr0,ip=dhcp -unprivileged 1 
-rootfs local:8
+ echo 'TurnKey nextcloud password: gNE05GvOiBQ='
+ msg 'Starting LXC Container...'
+ local 'TEXT=Starting LXC Container...'
+ echo -e 'Starting LXC Container...'
Starting LXC Container...
+ pct start 105
+ sleep 5
+ set +e
+ max_attempts=5
+ attempt=1
+ IP=
+ [[ 1 -le 5 ]]
++ pct exec 105 ip a show dev eth0
++ grep -oP 'inet \K[^/]+'
+ IP=
++ EXIT=1
++ LINE=212
++ error_exit
++ trap - ERR
++ local 'DEFAULT=Unknown failure occured.'
++ local 'REASON=\e[97mUnknown failure occured.\e[39m'
++ local 'FLAG=\e[91m[ERROR] ***@***.***'
++ msg '\e[91m[ERROR] ***@***.*** \e[97mUnknown failure occured.\e[39m'
++ local 'TEXT=\e[91m[ERROR] ***@***.*** \e[97mUnknown failure 
occured.\e[39m'
++ echo -e '\e[91m[ERROR] ***@***.*** \e[97mUnknown failure 
occured.\e[39m'
[ERROR] ***@***.*** Unknown failure occured.
++ '[' '!' -z 105 ']'
++ cleanup_ctid
++ pct status 105
+++ pct status 105
+++ awk '{print $2}'
++ '[' running == running ']'
++ pct stop 105
++ pct destroy 105
++ exit 1
***@***.***:~#

------ Original Message ------
From: "tteckster" ***@***.***>
To: "tteck/Proxmox" ***@***.***>
Cc: "swilkey" ***@***.***>; "Comment" 
***@***.***>
Sent: 5/12/2023 8:52:53 AM
Subject: Re: [tteck/Proxmox] TurnKey LXC Template Scripts (Discussion 
#1742)


Use the set -x option

bash -c "set -x; $(wget -qLO - https://github.com/tteck/Proxmox/raw/main/turnkey/turnkey.sh)"

—
Reply to this email directly, view it on GitHub 
<https://github.com/tteck/Proxmox/discussions/1742#discussioncomment-7758086>, 
or unsubscribe 
<https://github.com/notifications/unsubscribe-auth/AB5UXTFLTXKC2JIEFSF7WPDYHZA3LAVCNFSM6AAAAAA4DDCVYWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TONJYGA4DM>.
You are receiving this because you commented.Message ID: 
***@***.***>

**Reply by tteck** [reply]
I've seen this before, checking for an IP address is throwing an error. I turned off all error checking for that snippet. Hopefully this resolves your issue.

**Comment by swilkey** [comment]
Thanks. The interesting thing I found was that it was getting the IP
address correctly, but it was just throwing an error when checking the
address.  Made me think that there must be some assumption made when
checking it that isn't being satisfied.
